packages/core/src/llm/gemini.ts

```diff
@@ -98,32 +203,137 @@ export class GeminiSession {
     }
     this.gemini = new GoogleGenerativeAI(options.apiKey);
   }
+
+  getGenerativeModel(metadata: GoogleModelParams): GoogleGenerativeModel {
+    return this.gemini.getGenerativeModel(metadata);
+  }
+
+  getResponseText(response: EnhancedGenerateContentResponse): string {
+    return response.text();
+  }
+
+  async *getChatStream(
+    result: GoogleStreamGenerateContentResult,
+  ): GeminiChatStreamResponse {
+    yield* streamConverter(result.stream, (response) => ({
+      delta: this.getResponseText(response),
+      raw: response,
+    }));
+  }
+
+  getCompletionStream(
+    result: GoogleStreamGenerateContentResult,
+  ): AsyncIterable<CompletionResponse> {
+    return streamConverter(result.stream, (response) => ({
+      text: this.getResponseText(response),
+      raw: response,
+    }));
+  }
+}
+
+/* To use Google's Vertex AI backend, it doesn't use api key authentication.
+ *
+ * To authenticate for local development:
```

Comment:
can you update https://github.com/run-llama/LlamaIndexTS/blob/main/apps/docs/docs/modules/llms/available_llms/gemini.md and add it there?

Reply:
will do!